components/lora/lora_task.c

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 PWrInSpace, Kuba
+#include "lora_task.h"
+
+#include "esp_log.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/portmacro.h"
+#include "freertos/projdefs.h"
+#include "freertos/semphr.h"
+#include "freertos/task.h"
+#include "freertos/timers.h"
+
+#define TAG "LORA_T"
+
+static struct {
+    lora_struct_t lora;
+    lora_task_process_rx_packet process_packet_fnc;
+    lora_task_get_tx_packet get_tx_packet_fnc;
+    lora_state_t lora_state;
+
+    SemaphoreHandle_t irq_notification;
+    TaskHandle_t task;
+    TimerHandle_t receive_window_timer;
+} gb;
+
+// move to commands -- start
+// static bool check_lora_dev_id(lora_dev_id dev_id) {
+//     if (dev_id == BORADCAST_DEV_ID) {
+//         return true;
+//     }
+//     // remove privilage mask and check id
+//     return (dev_id >> 1) == (gb.dev_id >> 1) ? true : false;
+// }
+
+// static bool check_dev_id_privilage_mode(lora_dev_id dev_id) {
+//     return (dev_id & PRIVILAGE_MASK) > 0 ? true : false;
+// }
+// move to commands -- stop
+
+static bool wait_until_irq(void) {
+    // return xSemaphoreTake(gb.irq_notification, portMAX_DELAY) == pdTRUE ? true : false;
+    return ulTaskNotifyTake(pdTRUE, portMAX_DELAY) == pdTRUE ? true : false;
+}
+
+void IRAM_ATTR lora_task_irq_notifi(void *arg) {
+    BaseType_t higher_priority_task_woken = pdFALSE;
+    // xSemaphoreGiveFromISR(gb.irq_notification, &higher_priority_task_woken);
+    vTaskNotifyGiveFromISR(gb.task, &higher_priority_task_woken);
+    if (higher_priority_task_woken == pdTRUE) {
+        portYIELD_FROM_ISR();
+    }
+}
+
+static void notify_end_of_rx_window(void) { xTaskNotifyGive(gb.task); }
+
+static void lora_change_state_to_receive() {
+    lora_map_d0_interrupt(&gb.lora, LORA_IRQ_D0_RXDONE);
```

Comment:
w sumie jesli masz juz lora_state to moglbys go sprawdzac na poczatku funkcji change_state - zeby nie marnowac czasu na operacje jak juz jest dobry stan

dodatkowo imp lepiej zrobic jedna funkcje z parametrem lora_state_t na zmiane jej stanu i ewentualnie inline lora_change_state_to_receive, ktore by ja odpowiednio wywolywalo

Reply:
w sumie to ja nawet tego stanu za bardzo nie używam xD